situp-core/src/main/java/com/amazon/situp/pipeline/PipelineConnector.java

```diff
@@ -11,22 +11,30 @@
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicBoolean;
 
+import static java.lang.String.format;
+
 /**
  * PipelineConnector is a special type of Plugin which connects two pipelines acting both as Sink and Source.
- * TODO Add connecting pipeline details for better log messaging
  *
  * @param <T>
  */
 public final class PipelineConnector<T extends Record<?>> implements Source<T>, Sink<T> {
     private static final Logger LOG = LoggerFactory.getLogger(PipelineConnector.class);
     private static final int DEFAULT_WRITE_TIMEOUT = Integer.MAX_VALUE;
+    private String sourcePipelineName; //name of the pipeline for which this connector acts as source
+    private String sinkPipelineName; //name of the pipeline for which this connector acts as sink
     private Buffer<T> buffer;
     private AtomicBoolean isStopRequested;
 
     public PipelineConnector() {
         isStopRequested = new AtomicBoolean(false);
     }
 
+    public PipelineConnector(final String sinkPipelineName) {
+        this();
+        this.sinkPipelineName = sinkPipelineName;
+    }
+
```

Comment:
I guess this is fine for now but this is not a great way to use constructors in Java. Particularly the logic of this class heavily tied with PipelineParser. We should probably clean this up later, I will create a low prioirty github issue.